js/coinsbit.js

```diff
@@ -0,0 +1,1070 @@
+        //     success: true,
+        //     message: '',
+        //     result: {
+        //       offset: '0',
+        //       limit: '50',
+        //       records: [{
+        //         time: '1640258718.402',
+        //         fee: '0.01572657',
+        //         price: '3931.64271609',
+        //         amount: '0.002',
+        //         id: '1595921',
+        //         dealOrderId: '8631729321',  //<--- this is different than passed order-id
+        //         role: '2',
+        //         deal: '7.86328543'
+        //       },
+        //       ...
+        //     ]},
+        //     code: '200'
+        // }
+        const result = this.safeValue (response, 'result', {});
+        const order = this.safeValue (result, 'records');
+        if (!order) {
+            throw new OrderNotFound (this.id + ' fetchOrderTrades() could not find any trade data for order-id ' + id);
+        }
+        return this.parseTrade ({ 'id': id, 'order': order }, undefined);
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let currencyId = this.safeString (params, 'currency');
+        if (currencyId === undefined) {
+            const code = this.safeString (params, 'code');
+            if (code !== undefined) {
+                params = this.omit (params, 'code');
+                const currency = this.currency (code);
+                currencyId = currency['id'];
+            }
+        }
+        let method = 'privatePostAccountBalances';
+        const request = {};
+        if (currencyId !== undefined) {
+            request['currency'] = currencyId;
+            method = 'privatePostAccountBalance';
+        }
+        const response = await this[method] (this.extend (request, params));
+        // {
+        //     success: true,
+        //     message: '',
+        //     result: {
+        //       BTC: { available: '0', freeze: '0' },
+        //       USDT: { available: '0', freeze: '0' },
+        //       ...
+        const data = this.safeValue (response, 'result', {});
+        const result = {
+            'info': response,
+            'timestamp': undefined,
+            'datetime': undefined,
+        };
+        const currencyIds = Object.keys (data);
+        for (let i = 0; i < currencyIds.length; i++) {
+            const currencyId = currencyIds[i];
+            const code = this.safeCurrencyCode (currencyId);
+            const balance = this.safeValue (data, currencyId, {});
+            const account = this.account ();
+            account['free'] = this.safeString (balance, 'available');
+            account['used'] = this.safeString (balance, 'freeze');
+            result[code] = account;
+        }
+        return this.safeBalance (result);
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api];
```

Comment:
We have to implodeHostname here, it always goes in pair with `hostname` in `describe()` in all exchanges.